integration-tests/src/test/java/oracle/weblogic/kubernetes/ItLiftAndShiftFromOnPremDomain.java

```diff
@@ -0,0 +1,469 @@
+    logger.info("Push the image {0} to Docker repo", WKO_IMAGE_NAME);
+    dockerLoginAndPushImageToRegistry(WKO_IMAGE_NAME);
+
+    // Namespace and password needs to be updated in Create_k8s_secrets.sh
+    updateCreateSecretsFile();
+
+    // Namespace, domainHome, domainHomeSourceType, imageName and modelHome needs to be updated in wko-domain.yaml
+    updateDomainYamlFile();
+
+    // run create_k8s_secrets.sh that discoverDomain created to create necessary secrets
+    CommandParams params = new CommandParams().defaults();
+    params.command("sh "
+        + Paths.get(LIFT_AND_SHIFT_WORK_DIR, "/u01/", DISCOVER_DOMAIN_OUTPUT_DIR, "/create_k8s_secrets.sh").toString());
+
+    logger.info("Run create_k8s_secrets.sh to create secrets");
+    boolean result = Command.withParams(params).execute();
+    assertTrue(result, "Failed to create secrets");
+
+    logger.info("Run kubectl to create the domain");
+    params = new CommandParams().defaults();
+    params.command("kubectl apply -f "
+        + Paths.get(LIFT_AND_SHIFT_WORK_DIR, "/u01/", DISCOVER_DOMAIN_OUTPUT_DIR + "/" + WKO_DOMAIN_YAML).toString());
+
+    result = Command.withParams(params).execute();
+    assertTrue(result, "Failed to create domain custom resource");
+
+    // wait for the domain to exist
+    logger.info("Checking for domain custom resource in namespace {0}", domainNamespace);
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for domain {0} to be created in namespace {1} "
+                    + "(elapsed time {2}ms, remaining time {3}ms)",
+                domainUid,
+                domainNamespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(domainExists(domainUid, DOMAIN_VERSION, domainNamespace));
+
+    // verify the admin server service and pod is created
+    checkPodReadyAndServiceExists(adminServerPodName, domainUid, domainNamespace);
+
+    // verify managed server services created and pods are ready
+    for (int i = 1; i <= replicaCount; i++) {
+      checkPodReadyAndServiceExists(managedServerPrefix + i, domainUid, domainNamespace);
+    }
+
+    // create ingress rules with path routing for Traefik
+    createTraefikIngressRoutingRules(domainNamespace);
+
+    int traefikNodePort = getServiceNodePort(traefikNamespace, traefikHelmParams.getReleaseName(), "web");
+    assertTrue(traefikNodePort != -1,
+        "Could not get the default external service node port");
+    logger.info("Found the Traefik service nodePort {0}", traefikNodePort);
+    logger.info("The K8S_NODEPORT_HOST is {0}", K8S_NODEPORT_HOST);
+
+    String curlString = String.format("curl -v --show-error --noproxy '*' "
+            + "http://%s:%s/opdemo/?dsName=testDatasource",
+        K8S_NODEPORT_HOST, traefikNodePort);
+
+    ExecResult execResult = null;
+    logger.info("curl command {0}", curlString);
+
+    try {
```

Comment:
Use the withStandardRetryPolicy  API instead of using sleep

Reply:
Done